### ISSUE TITLE
tslint:disable-next-line difficult to apply to object-literal-sort-keys

### ISSUE THREAD

**Issue by rhys-vdw** [palantir/tslint#3597]
### Bug Report

- __TSLint version__: 5.8.0
- __TypeScript version__: 2.6.1
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
// tslint:disable-next-line:object-literal-sort-keys
 const UiBreakpoints = {
  xxsmall: 0,
  xsmall: 450,
  small: 700,
  large: 800,
}
```

with `tslint.json` configuration:

```json
{
  "extends": [
    "tslint:recommended",
    "tslint-eslint-rules",
    "tslint-react"
  ],
  "defaultSeverity": "error",
  "rules": {
    "array-bracket-spacing": [true, "never"],
    "curly": [true, "ignore-same-line"],
    "indent": [true, "spaces", 2],
    "interface-name": false,
    "jsx-curly-spacing": [true, "never"],
    "jsx-no-multiline-js": false,
    "jsx-boolean-value": [true, "never"],
    "max-line-length": [true, 120],
    "member-access": [true, "no-public"],
    "no-console": [true, "warn", "error"],
    "no-constant-condition": true,
    "no-debugger": true,
    "no-unused-variable": [true, "check-parameters", { "ignorePattern": "^_" }],
    "no-var-keyword": true,
    "no-var-requires": false,
    "object-curly-spacing": [true, "always"],
    "ordered-imports": false,
    "prefer-const": true,
    "quotemark": [true, "double", "jsx-double", "avoid-escape"],
    "semicolon": [true, "never"],
    "trailing-comma": [true, {
      "multiline": {
        "objects": "always",
        "arrays": "always",
        "functions": "ignore",
        "typeLiterals": "always",
        "imports": "always",
        "exports": "always"
      },
      "singleline": "never"
    }],
    "variable-name": {
      "options": [
        "allow-leading-underscore",
        "ban-keywords",
        "check-format",
        "allow-pascal-case"
      ]
    }
  },
  "linterOptions": {
    "exclude": ["./node_modules/**/*"]
  }
}
```

#### Actual behavior

> The key 'xsmall' is not sorted alphabetically (object-literal-sort-keys)

#### Expected behavior

Rule disabled for entire object definition.

**Comment by rhys-vdw** [comment]
Ah, no sooner did I post it did I see the example on https://palantir.github.io/tslint/usage/rule-flags/. Still it would be nice to have this behaviour changed. Feel free to close the issue.

**Comment by adidahiya** [comment]
Do any other linters work like this? I'm not sure this proposed behavior will be intuitive. For example, what if I want to only disable the sorting check for one of the fields in a literal? It will look confusing to see a `disable-next-line` flag do something different based on the line where it's placed.

**Comment by rhys-vdw** [comment]
> For example, what if I want to only disable the sorting check for one of the fields in a literal?

Fair point.
